Compilation warning

```
./lib/okon.cpp: In function 'okon_prepare_result okon_prepare(const char*, const char*, const char*, okon_prepare_progress_callback_t, void*)':
./lib/okon.cpp:23:5: warning: control reaches end of non-void function [-Wreturn-type]
   23 |   }();
      |     ^
```

--- a/include/okon/okon.h
+++ b/include/okon/okon.h
@@ -10,7 +10,8 @@ enum okon_prepare_result
   okon_prepare_result_could_not_open_input_file,         //!< Issue while opening input file.
   okon_prepare_result_could_not_open_intermediate_files, //!< Issue while creating intermediate
                                                          //!< files.
-  okon_prepare_result_could_not_open_output              //!< Issue while creating output file.
+  okon_prepare_result_could_not_open_output,             //!< Issue while creating output file.
+  okon_prepare_result_unspecified_failure                //!< Unspecified failure occurred
 };
 
 enum okon_prepare_progress_special_value

--- a/lib/okon.cpp
+++ b/lib/okon.cpp
@@ -36,6 +36,8 @@ okon_prepare_result okon_prepare(const char* input_db_file_path, const char* wor
     case okon::preparer::result ::could_not_open_output:
       return okon_prepare_result ::okon_prepare_result_could_not_open_output;
   }
+
+  return okon_prepare_result ::okon_prepare_result_unspecified_failure;
 }
 
 okon_exists_result okon_exists_text(const char* sha1, const char* processed_file_path)

--- a/lib/okon.cpp
+++ b/lib/okon.cpp
@@ -48,7 +48,8 @@ okon_exists_result okon_exists_text(const char* sha1, const char* processed_file
 
 okon_exists_result okon_exists_binary(const void* sha1, const char* processed_file_path)
 {
-  okon::fstream_wrapper file{ processed_file_path };
+  okon::fstream_wrapper file{ processed_file_path,
+                              std::ios::in | std::ios::out | std::ios::binary };
 
   if (!file.is_open()) {
     return okon_exists_result::okon_prepare_result_could_not_open_file;

